src/decoder/types.ts

```diff
@@ -0,0 +1,89 @@
+import { bufferToHex, bufferToInt, addHexPrefix } from 'ethereumjs-util';
+import { ensure } from '../util/ensure';
+
+enum Primitive {
+    Address,
+    Bool,
+    String,
+    Bytes,
+    Int,
+}
+
+function formatBool(value: Buffer): boolean {
```

Comment:
These all look right but should we be implementing our own primitives like this? Is there any library we can use that might be a bit cleaner?

Reply:
Unfortunately not. The type strings are solidity types (uint256, uint256[], etc.), and I wasn't able to find a library that does this for us. ethereumjs-abi parses and decodes the types, but decides to convert them all to `Buffer` unfortunately (except address), so the point of this file is just to convert to the native JS types that we know and love. If you have an alternative idea, let me know - I'm not too mad about having to do it ourselves here, though. For example if we decide that we want to change the way the formatting looks (i.e. remove 0x prefix, add EIP-55 casing, etc.) - we have the control and power to do it ourselves this way

Reply:
That's alright with me - just wanted to make sure there was some logic for it :)